spk/sonarr/src/service-setup.sh

```diff
@@ -1,104 +1,89 @@
 HOME_DIR="${SYNOPKG_PKGVAR}"
 CONFIG_DIR="${SYNOPKG_PKGVAR}/.config"
+SONARR_CONFIG_DIR="${CONFIG_DIR}/Sonarr"
+PID_FILE="${SONARR_CONFIG_DIR}/sonarr.pid"
 
-# Sonarr v2 -> v3 compatibility:
-if [ -f "${SYNOPKG_PKGDEST}/share/NzbDrone/NzbDrone.exe" ]; then
-    # v2 installed
-    SONARR="${SYNOPKG_PKGDEST}/share/NzbDrone/NzbDrone.exe"
-    PID_FILE="${CONFIG_DIR}/NzbDrone/nzbdrone.pid"
-else
-    # v3 installed
-    SONARR="${SYNOPKG_PKGDEST}/share/Sonarr/Sonarr.exe"
-    PID_FILE="${CONFIG_DIR}/Sonarr/sonarr.pid"
-fi
-
-# Allow correct Sonarr SPK version checking (v2 or v3)
-if [ -f "${SYNOPKG_PKGINST_TEMP_DIR}/share/NzbDrone/NzbDrone.exe" ]; then
-    # v2 SPK
-    SPK_SONARR="${SYNOPKG_PKGINST_TEMP_DIR}/share/NzbDrone/NzbDrone.exe"
-else
-    # v3 SPK
-    SPK_SONARR="${SYNOPKG_PKGINST_TEMP_DIR}/share/Sonarr/Sonarr.exe"
-fi
-
+LEGACY_SPK_NAME="nzbdrone"
+LEGACY_SYNOPKG_PKGDEST="/var/packages/${LEGACY_SPK_NAME}/target"
+LEGACY_SYNOPKG_PKGVAR="/var/packages/${LEGACY_SPK_NAME}/var"
+LEGACY_CONFIG_DIR="${LEGACY_SYNOPKG_PKGVAR}/.config"
 # Some have it stored in the root of package
-LEGACY_CONFIG_DIR="${SYNOPKG_PKGDEST}/.config"
-
-# workaround for mono bug with armv5 (https://github.com/mono/mono/issues/12537)
-if [ "$SYNOPKG_DSM_ARCH" == "88f6281" -o "$SYNOPKG_DSM_ARCH" == "88f6282" ]; then
-    MONO="MONO_ENV_OPTIONS='-O=-aot,-float32' ${MONO}"
-fi
+LEGACY_OLD_CONFIG_DIR="${LEGACY_SYNOPKG_PKGDEST}/.config"
 
+# for DSM < 7 only:
 GROUP="sc-download"
 LEGACY_GROUP="sc-media"
 
-SERVICE_COMMAND="env PATH=${MONO_PATH}:${PATH} HOME=${HOME_DIR} LD_LIBRARY_PATH=${SYNOPKG_PKGDEST}/lib ${MONO} ${SONARR}"
+SERVICE_COMMAND="env HOME=${HOME_DIR} LD_LIBRARY_PATH=${SYNOPKG_PKGDEST}/lib ${SONARR}"
 SVC_BACKGROUND=y
 SVC_WAIT_TIMEOUT=90
 
-service_postinst ()
+validate_prereplace ()
 {
-    mkdir -p ${CONFIG_DIR}
-    set_unix_permissions "${CONFIG_DIR}"
+    # use install_log to write to installer log file.
+    install_log "validate_prereplace ${SYNOPKG_PKG_STATUS}"
+
+    # check if the installed distribution is a legacy version
+    if [ -f "${LEGACY_SYNOPKG_PKGDEST}/share/NzbDrone/NzbDrone.exe" ]; then
+        # v2 installed
+        exit 1
+    elif [ -f "${LEGACY_SYNOPKG_PKGDEST}/share/Sonarr/Sonarr.exe" ]; then
+        # v3 installed
+        exit 0
+    fi
 }
 
-service_preupgrade ()
+service_prereplace ()
 {
-    # We have to account for legacy folder in the root
-    # It should go, after the upgrade, into /var/.config/
-    # The /var/ folder gets automatically copied by service-installer after this
-    if [ -d "${LEGACY_CONFIG_DIR}" ]; then
-        echo "Moving ${LEGACY_CONFIG_DIR} to ${CONFIG_DIR}"
-        mv ${LEGACY_CONFIG_DIR} ${CONFIG_DIR} >> ${INST_LOG} 2>&1
-    fi
-    if [ ! -d ${CONFIG_DIR} ]; then
-        # Create, in case it's missing for some reason
-        mkdir -p ${CONFIG_DIR} >> ${INST_LOG} 2>&1
+    # use echo to write to the installer log file.
+    echo "service_prereplace ${SYNOPKG_PKG_STATUS}"
+    
```

Comment:
In the generic service installer we already have
```
log_step "prereplace"
```

So there is already an entry in install log for the `prereplace` step.


As it might be relevant for all packages, I propose to add  `${SYNOPKG_PKG_STATUS}` to the install log in `log_step` function defined in `mk/spksrc.service.installer.functions` like:

```
log_step ()
{
    install_log "===> Step $1. STATUS=${SYNOPKG_PKG_STATUS} USER=${USER} GROUP=${GROUP} SHARE_PATH=${SHARE_PATH}"
}
```